lib/vrf.c

```diff
@@ -497,6 +497,16 @@ void vrf_init(int (*create)(struct vrf *), int (*enable)(struct vrf *),
 		strlcpy(default_vrf->data.l.netns_name,
 			VRF_DEFAULT_NAME, NS_NAMSIZ);
 
+	if (vrf_is_backend_netns()) {
+		struct ns *ns;
+
+		strlcpy(default_vrf->data.l.netns_name,
+			VRF_DEFAULT_NAME, NS_NAMSIZ);
+		ns = ns_lookup(ns_get_default_id());
+		ns->vrf_ctxt = (void *)default_vrf;
+		default_vrf->ns_ctxt = (void *)ns;
```

Comment:
additional nit: casting to/from void pointers is unnecessary in modern C. I'd prefer to remove the explicit cast to simplify the code.